Makefile

```diff
@@ -199,112 +182,128 @@ github-token.txt:
-# the appropriate devkit image for your system
-.PHONY: devkit-arm64
-devkit-arm64:
-devkit-arm64: buildx github-token.txt
-		docker buildx build \
-		-t docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-arm64 \
-		--build-arg USER_ID=$(UID) \
-		--build-arg GROUP_ID=$(GID) \
-		--build-arg USER_NAME=$(USER_NAME) \
-		--build-arg GROUP_NAME=$(GROUP_NAME) \
-		--build-arg DOCKER_GID=$(DOCKER_SOCKET_GID) \
-		--build-arg BUILDARCH=arm64 \
-		--platform linux/arm64 \
-		--file $(REPO_ROOT_DIR)/Dockerfile.devkit \
-		--secret id=githubtoken,src=github-token.txt \
-		--provenance=false  \
-		--push \
-		$(REPO_ROOT_DIR)
+	 docker run --privileged --rm tonistiigi/binfmt --install all &>/dev/null || true
 
-.PHONY: devkit-amd64
-devkit-amd64:
-devkit-amd64: buildx github-token.txt
-		docker buildx build \
-		-t docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-amd64 \
-		--build-arg USER_ID=$(UID) \
-		--build-arg GROUP_ID=$(GID) \
-		--build-arg USER_NAME=$(USER_NAME) \
-		--build-arg GROUP_NAME=$(GROUP_NAME) \
-		--build-arg DOCKER_GID=$(DOCKER_SOCKET_GID) \
-		--build-arg BUILDARCH=amd64 \
-		--platform linux/amd64 \
-		--file $(REPO_ROOT_DIR)/Dockerfile.devkit \
-		--secret id=githubtoken,src=github-token.txt \
-		--provenance=false  \
-		--push \
-		$(REPO_ROOT_DIR)
+###### Devkit container image
+DEVKIT_IMAGE_DOCKERFILE ?= Dockerfile.devkit
+DEVKIT_IMAGE_NAME ?= mesosphere/konvoy-image-builder-devkit
+DEVKIT_IMAGE_TAG ?= $(shell cat ${DEVKIT_IMAGE_DOCKERFILE} requirements.txt requirements-devkit.txt  | sha256sum | cut -d" " -f 1)
 
-.PHONY: docker-build-amd64
-docker-build-amd64: BUILDARCH=amd64
-docker-build-amd64: devkit-amd64 konvoy-image-amd64
-	docker buildx build \
-		--file $(REPO_ROOT_DIR)/Dockerfile \
-		--build-arg BUILDARCH=amd64 \
-		--platform linux/amd64 \
-		--build-arg BASE=docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-amd64 \
-		--tag=$(DOCKER_REPOSITORY):$(REPO_REV)-amd64 \
-		--pull \
-		--provenance=false \
-		--push \
-		$(REPO_ROOT_DIR)
+.PHONY: devkit-image
+## first tries to pull an image, if doesn't exist build and push the image
+devkit-image:
+	$(call print-target)
+	docker image inspect $(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG) &>/dev/null || \
+	docker pull $(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG) || \
+	$(MAKE) devkit-image-amd64 devkit-image-arm64
 
-.PHONY: docker-build-arm64
-docker-build-arm64: BUILDARCH=arm64
-docker-build-arm64: devkit-arm64 konvoy-image-arm64
-	docker buildx build \
-		--file $(REPO_ROOT_DIR)/Dockerfile \
-		--build-arg BUILDARCH=arm64 \
-		--platform linux/arm64 \
-		--build-arg BASE=docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-arm64 \
-		--tag=$(DOCKER_REPOSITORY):$(REPO_REV)-arm64 \
-		--pull \
-		--push \
-		--provenance=false  \
-		$(REPO_ROOT_DIR)
+.PHONY: devkit-image-amd64
+devkit-image-amd64:
+	$(MAKE) devkit-image-build-push BUILDARCH=amd64
 
+.PHONY: devkit-image-arm64
+devkit-image-arm64:
+	$(MAKE) devkit-image-build-push BUILDARCH=arm64
```

Comment:
i don't want to use recursive calls of $(MAKE) when we can specify dependencies

Reply:
Maybe we should abandon the Makefile completely for this sort of thing :thinking: 

>  However when specified as dependency, the variable BUILDARCH was not overriding default value.

I was having trouble with that as well

Reply:
Agree. we can do some POC later for the alternatives. This Makefile is still a mess. We can either make it a bit more simpler or move to mage.